zppy/utils.py

```diff
@@ -140,9 +143,135 @@ def getComponent(input_files):
+                        # by the time we get to the current script
+                        # (since __main__.py runs the tasks in dependency order).
+                        # So, continue on to next dependency.
+                        self.dependencies_in_bundle_file.add(
+                            dependFile
+                        )  # Useful for debugging
+                        break
+                else:
+                    # If we're in this block, then we did not `break` from the for-loop.
+                    # So, the dependency is not in this bundle.
+                    # So, the dependency will need to be finished before we can run this bundle.
+                    # We must pass this information on via dependencies_not_in_bundle_file
+                    self.dependencies_not_in_bundle_file.add(dependFile)
+
+    def add_script(self, scriptFile):
+        # Add script file to bundle
+        with open(self.bundle_file, "a") as f:
+            # Add scriptFile to the list of scripts to run
+            # The header in scriptFile will simply be ignored
+            file_name_only = os.path.split(scriptFile)[-1]
+            # Put blank line between each script call
+            f.write("\n")
+            # Allow script to be run
+            f.write(f"chmod 760 {file_name_only}\n")  # Default is 660
+            # Run script
+            f.write(f"./{file_name_only}\n")
+
+    # Useful for debugging
+    def display_dependencies(self):
+        print(f"Displaying dependencies for {self.bundle_name}")
+        print("dependencies_in_bundle_file:")
+        for dependency in self.dependencies_in_bundle_file:
+            d = os.path.split(dependency)[-1]
+            print(f"  {d}")
+        else:
+            # If we're in this block, then the list is empty.
+            print("  None")
+        print("dependencies_not_in_bundle_file:")
+        for dependency in self.dependencies_not_in_bundle_file:
+            d = os.path.split(dependency)[-1]
+            print(f"  {d}")
+        else:
+            # If we're in this block, then the list is empty.
+            print("  None")
+
+
+def handle_bundles(c, scriptFile, export, dependFiles=[], existing_bundles=[]):
+    bundle_name = c["bundle"]
+    if bundle_name == "":
+        return existing_bundles
+    for b in existing_bundles:
+        if b.bundle_name == bundle_name:
+            # This bundle already exists
+            bundle = b
+            break
+    else:
+        # If we're in this block, then we did not `break` from the for-loop.
+        # So, the bundle does not already exist
+        bundle = Bundle(c)
+        bundle.create_header(scriptFile)
+        existing_bundles.append(bundle)
+    bundle.handle_dependencies(dependFiles)
+    bundle.add_script(scriptFile)
+    if export == "ALL":
+        # If one task requires export="ALL", then the bundle script will need it as well
+        bundle.export = export
```

Comment:
Since we're only launching one job rather than many, we have to make a decision on the value for `export`. I decided that if *any* task required `export="ALL"`, then the whole bundle would.